Float.Core/UX/CoordinatorParent.cs

```diff
@@ -117,6 +123,33 @@ public override string ToString()
             return $"[{GetType()}, Children: {string.Join(",", childCoordinators)}]";
         }
 
+        /// <summary>
+        /// Handles the finish request.
+        /// </summary>
+        /// <param name="coordinator">The calling coordinator.</param>
+        /// <param name="eventArgs">The event args.</param>
+        /// <returns>A value indicating whether this finished.</returns>
+        public override bool HandleFinishRequested(ICoordinator coordinator, EventArgs eventArgs)
+        {
+            var didFinish = true;
+
+            waitingToFinishEventArgs = eventArgs;
+
+            foreach (var eachChild in ChildCoordinators)
+            {
+                if (eachChild is Coordinator childCoordinator)
+                {
+                    var finished = childCoordinator.HandleFinishRequested(childCoordinator, eventArgs);
+                    if (!finished)
+                    {
+                        didFinish = false;
+                    }
+                }
+            }
+
```

Comment:
@EBusch We're probably missing a call to the base implementation here. When you add that, you may have an issue with `Finish` getting called twice:

`Coordinator.HandleNavigation` _may_ call it and so will `CoordinatorParent.HandleChildFinish` -- it is not currently clear to me the best way to address that.